src/test/java/com/wikia/webdriver/common/testnglisteners/BrowserAndTestEventListener.java

```diff
@@ -0,0 +1,265 @@
+                                   .getWikiaDomain())));
+      }
+    }
+
+    if (TestContext.isFirstLoad()) {
+      User user = null;
+      TestContext.setFirstLoad(false);
+
+      if (declaringClass.isAnnotationPresent(Execute.class)) {
+        user = declaringClass.getAnnotation(Execute.class).asUser();
+      }
+
+      if (method.isAnnotationPresent(Execute.class)) {
+        user = method.getAnnotation(Execute.class).asUser();
+      }
+
+      if (user != null && user != User.ANONYMOUS) {
+        // log in, make sure user is logged in and flow is on the requested url
+        new WikiBasePageObject().loginAs(user);
+      }
+      NetworkTrafficInterceptor
+          networkTrafficInterceptor =
+          DriverProvider.getActiveDriver().getProxy();
+      if (networkTrafficInterceptor != null && Configuration.getForceHttps()) {
+        networkTrafficInterceptor.startIntercepting();
+      }
+    }
+
+    Log.logJSError();
+  }
+
+  @Override
+  public void beforeFindBy(By by, WebElement element, WebDriver driver) {
+    lastFindBy = by;
+    SelectorStack.write(by);
+    if (element != null) {
+      SelectorStack.contextWrite();
+    }
+  }
+
+  @Override
+  public void beforeClickOn(WebElement element, WebDriver driver) {
+    Log.logJSError();
+  }
+
+  @Override
+  public void afterClickOn(WebElement element, WebDriver driver) {
+    Log.info("click", lastFindBy.toString());
+  }
+
+  @Override
+  public void afterChangeValueOf(WebElement element, WebDriver driver, CharSequence[] keysToSend) {
+    Log.info("ChangeValueOfField", lastFindBy.toString());
+  }
+
+  @Override
+  public void onTestStart(ITestResult result) {
+    Log.clearLogStack();
+    String testName = result.getName();
+    String className = result.getTestClass().getName();
+    System.out.println(className + " " + testName);
+  }
+
+  @Override
+  public void onTestSuccess(ITestResult result) {
+    Log.stop();
+  }
+
+  @Override
+  public void onTestFailure(ITestResult result) {
+    driver = DriverProvider.getActiveDriver();
+
+    if ("true".equals(Configuration.getLogEnabled())) {
+      Log.logError("Test Failed", result.getThrowable());
+      Log.logJSError();
+      Log.stop();
+    }
+  }
+
+  @Override
+  public void onTestSkipped(ITestResult result) {
+    if (!Log.isTestStarted()) {
+      Log.startTest(result.getMethod().getConstructorOrMethod().getMethod());
+    }
+    if (result.getMethod().getConstructorOrMethod().getMethod()
+        .isAnnotationPresent(DontRun.class)) {
+      Log.ok("Test SKIPPED", "this test is not supported in this environment");
+      result.setStatus(ITestResult.SUCCESS);
+      onTestSuccess(result);
+    } else {
+      result.setStatus(ITestResult.FAILURE);
+      result.setThrowable(new SkipException("TEST SKIPPED"));
+      onTestFailure(result);
+    }
+    if (Log.isTestStarted()) {
+      Log.stop();
+    }
+  }
+
+  @Override
+  public void onTestFailedButWithinSuccessPercentage(ITestResult result) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Add a nested comment explaining why this method is empty, throw an UnsupportedOperationException or complete the implementation. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar-s4/coding_rules#rule_key=squid%3AS1186)